pkg/cloud/aliyunprovider.go

```diff
@@ -322,59 +370,58 @@ func (alibaba *Alibaba) DownloadPricingData() error {
 		alibaba.Pricing[lookupKey] = pricingObj
 	}
 
-	// TO-DO: PV pricing
-	// //get pvList ultimately from Alibaba cloud provider and resemble data from the pvtype to
-	// // Hardcodedk8sNodeDiskStruct
-	// pvList := alibaba.Clientset.GetAllPersistentVolumes()
-
-	// pvList := []*Hardcodedk8sNodeDiskStruct{}
-	// pvList = append(pvList, &Hardcodedk8sNodeDiskStruct{
-	// 	DiskType:         "data",
-	// 	DiskCategory:     "cloud",
-	// 	PerformanceLevel: "",
-	// 	RegionID:         "cn-hangzhou",
-	// 	PriceUnit:        "Hour",
-	// 	SizeInGiB:        60,
-	// 	ProviderID:       "Ali-XXX-pv-01",
-	// }, &Hardcodedk8sNodeDiskStruct{
-	// 	DiskType:         "data",
-	// 	DiskCategory:     "cloud",
-	// 	PerformanceLevel: "P1",
-	// 	RegionID:         "cn-hangzhou",
-	// 	PriceUnit:        "Hour",
-	// 	SizeInGiB:        40,
-	// 	ProviderID:       "Ali-XXX-pv-01",
-	// })
-
-	// for _, pv := range pvList {
-	// 	if client, ok = alibaba.clients[pv.RegionID]; !ok {
-	// 		client, err = sdk.NewClientWithAccessKey(pv.RegionID, aak.AccessKeyId, aak.AccessKeySecret)
-	// 		if err != nil {
-	// 			return fmt.Errorf("access key provided does not have access to location %s", pv.RegionID)
-	// 		}
-	// 		alibaba.clients[pv.RegionID] = client
-	// 	}
-	// 	signer = signers.NewAccessKeySigner(aak)
-	// 	pricingObj, err = processDescribePriceAndCreateAlibabaPricing(client, pv, signer)
-	// 	lookupKey, err = determineKeyForPricing(pv)
-	// 	if err != nil {
-	// 		return err
-	// 	}
-	// 	alibaba.Pricing[lookupKey] = pricingObj
-	// }
-	// log.Infof("Length of pricing is %d", len(alibaba.Pricing))
-	// log.Infof("random value is %v", alibaba.Pricing[lookupKey])
+	// set the first occurance of region from the node
+	i := 0
+	for alibaba.clusterRegion == "" && i < len(nodeList) {
+		regionID, ok := nodeList[i].Labels["topology.kubernetes.io/region"]
+		if ok {
+			alibaba.clusterRegion = regionID
+		} else {
+			i += 1
+		}
+	}
+
+	// PV pricing for only Cloud Disk for now.
+	// TO-DO: Support both NAS(Network Attached storage) and OSS(Object Storage Service) type PVs
+
+	pvList := alibaba.Clientset.GetAllPersistentVolumes()
+
+	for _, pv := range pvList {
+		pricingObj := &AlibabaPricing{}
+		// Note for PR: Region ID defaulted to first occurance of node region for now, need to know the implication here!
+		// Can Pvs be in seperate Region than the Cluster? Is there ever a Multi Region k8s cluster?
```

Comment:
Thanks for calling this out in a comment! I think most cloud providers support multi-regional/multi-zonal clusters, i.e. nodes in different regions. If Alibaba supports that, then I suspect that there could be PVs in different regions but all in the same cluster.

Have you checked if the PVs/PVCs in the cluster have a `topology` annotation of any kind?

Reply:
This is definitely most tricking PR comment and closest thing you are suggesting via PV describe is in node affinity part of it but it gives affinity zone, I would have to programmatically align it with a region. What do you think about that?
Like in the below scenario us-east-1a is the zone i'll have to make sure it's region is us-east-1.
![Screen Shot 2022-11-23 at 3 16 27 PM](https://user-images.githubusercontent.com/11470561/203662123-8235d4c2-8d5e-4b8f-bec3-a1d88b5335c7.png)

Reply:
as an FYI 
`apiVersion: v1
kind: PersistentVolume
metadata:
  name: yaml-pv
  labels:
    alicloud-pvname: static-disk-yaml-pv
spec:
  capacity:
    storage: 25Gi
  accessModes:
    - ReadWriteOnce
  persistentVolumeReclaimPolicy: Retain
  csi:
    driver: diskplugin.csi.alibabacloud.com
    volumeHandle: "d-0xi2gxpy6za03i2cffff"
  nodeAffinity:
    required:
      nodeSelectorTerms:
      - matchExpressions:
        - key: topology.diskplugin.csi.alibabacloud.com/zone
          operator: In
          values:
          - "us-east-1a"
`

I was only able to create pv with one node affinity. It doesnt allow me to affiliate a PV with multiple zone such as us-east-1a, us-east1b let alone multiple region. Can safely say first occurance of value and determining region based on that should be enough .